infra/build/functions/update_build_status.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Cloud function to request builds."""
+import logging
+import tempfile
+
+import google.auth
+from googleapiclient.discovery import build
+from google.cloud import ndb
+from google.cloud import storage
+
+import build_and_run_coverage
+import build_project
+import builds_status
+from datastore_entities import BuildsHistory
+
+
+# TODO: update function in builds_status.py after migration.
+def get_last_build(builds, project, build_tag_suffix):
+  """Returns build object for the last finished build of project."""
+  with ndb.Client().context():
```

Comment:
Use a top level ndb context in the entry point of your function.

Does this even work as-is when you have .get() on line 38?